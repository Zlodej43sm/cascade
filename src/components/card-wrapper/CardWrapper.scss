@use '../../styles/colors';
@use '../../styles/typography';
@use '../../styles/mixins';
@use '../../styles/sizes';

.container {
  @include mixins.themeBoxShadow(colors.$base-darkest);
  padding: sizes.$base-3 sizes.$base-3 sizes.$base-5;
  background-color: colors.$base-lightest;
  border-radius: 10px;
}

.header {
  display: flex;
  margin-bottom: sizes.$base-3;
}

.icon {
  font-size: 42px;
  margin-right: sizes.$base-3;
  flex: sizes.$base-4 0 0;
  text-align: center;
  position: relative;
  top: - sizes.$base;
}

.icon,
.subTitle {
  color: colors.$cascade-secondary;
}

.title {
  @include typography.title-md;
  color: colors.$cascade-primary;
}

.subTitle {
  @include typography.sub-title;
}

.content {
  overflow: auto;
}
