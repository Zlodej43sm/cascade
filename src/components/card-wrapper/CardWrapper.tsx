import React from 'react';

import styles from './CardWrapper.scss';

export type CardWrapperProps = {
  icon?: React.ReactNode
  title?: string
  subTitle?: string
  children?: React.ReactNode
  width?: number | string
  height?: number | string
};

function CardWrapper({ icon, title, subTitle, children, width, height }: CardWrapperProps) {
  return (
    <section className={styles.container} style={{ width, height }} data-testid="card-wrapper-container">
      <header className={styles.header}>
        {icon && (
          <div className={styles.icon} data-testid="card-wrapper-icon">
            <span>{icon}</span>
          </div>
        )}
        <div>
          {title && (
            <div className={styles.title} data-testid="card-wrapper-title">
              {title}
            </div>
          )}
          {subTitle && (
            <div className={styles.subTitle} data-testid="card-wrapper-subtitle">
              {subTitle}
            </div>
          )}
        </div>
      </header>
      <div className={styles.content}>
        {children}
      </div>
    </section>
  );
}

export default CardWrapper;
