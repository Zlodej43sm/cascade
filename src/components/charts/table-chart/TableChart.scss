@use '../../../styles/colors';
@use '../../../styles/typography';
@use '../../../styles/mixins';
@use '../../../styles/sizes';


.grid {
  display: grid;
  grid-template-columns: minmax(sizes.$base * 18, 1.5fr) repeat(var(--columns-count), minmax(sizes.$base * 10, 1fr)) sizes.$base-6;
  grid-gap: sizes.$base-3 sizes.$base;
  padding-top: sizes.$base-4;
}

.row {
  display: contents;

  &>:first-child {
    grid-column: 1;
  }
}

.header {
  @include typography.poppins(11px, 12px);
  align-self: center;
  text-align: center;
  color: colors.$base-grey;
  transform: rotate(-45deg);
  padding: 10% 0;

  &:first-child {
    grid-column: 2;
  }
}

.name {
  @include typography.poppins(14px, 21px, 500);
  align-self: center;
  color: colors.$base-dark;
  display: flex;
  align-items: center;
}

.avatar {
  display: block;
  width: 40px;
  height: 40px;
  object-fit: cover;
  margin-right: 12px;
  border-radius: 50%;
  border: 1.5px solid colors.$cascade-secondary;
  box-sizing: border-box;
}

.value {
  @include typography.poppins(10px, 15px, 500);
  height: sizes.$base-5;
  border-radius: 5px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.legend {
  grid-column: calc(var(--columns-count) + 2);
  grid-row-start: 2;
  grid-row-end: calc(var(--rows-count) + 2);
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
  flex-direction: column-reverse;
}

.color {
  @include typography.poppins(10px, 15px, 500);
  width: sizes.$base-4;
  height: sizes.$base-4;
  display: flex;
  align-items: center;
  justify-content: center;

  &:first-child {
    border-radius: 0 0 5px 5px;
  }

  &:last-child {
    border-radius: 5px 5px 0 0;
  }
}
