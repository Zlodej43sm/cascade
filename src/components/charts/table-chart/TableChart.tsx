/* eslint-disable react/no-array-index-key */
import { useMemo } from 'react';

import CardWrapper, { CardWrapperProps } from 'lib/components/card-wrapper/CardWrapper';

import styles from './TableChart.scss';

type RowData = {
  name: string,
  avatar?: string,
  data: number[],
};

type CellColorData = {
  text: string,
  background: string,
};

export type TableChartProps = {
  columns: string[],
  series: RowData[],
  colors?: CellColorData[] | Record<number, CellColorData>,
  getColors?: (value: number) => CellColorData,
};

export function TableChart({ columns, series, colors, getColors }: TableChartProps) {
  const colorsMap = useMemo(() => {
    if (colors) {
      return colors;
    }
    if (getColors) {
      const uniqueValues = Array.from(new Set(series.flatMap((s) => s.data))).sort();
      return uniqueValues.reduce<Record<number, CellColorData>>((acc, value) => {
        acc[value] = getColors(value);
        return acc;
      }, {});
    }
    return {};
  }, [series, colors, getColors]);

  return (
    <div
      className={styles.grid}
      style={{ '--columns-count': columns.length, '--rows-count': series.length } as React.CSSProperties}
      data-testid="table-chart"
    >
      <div className={styles.row}>
        {columns.map((text, index) => (
          <div key={index} className={styles.header}>{text}</div>
        ))}
      </div>
      {series.map(({ name, data, avatar }, index) => (
        <div key={index} className={styles.row}>
          <div className={styles.name}>
            {avatar && <img src={avatar} alt={name} className={styles.avatar} />}
            {name}
          </div>
          {data.map((value, i) => (
            <div
              key={i}
              className={styles.value}
              style={{ color: colorsMap[value]?.text, backgroundColor: colorsMap[value]?.background }}
            >
              {value}
            </div>
          ))}
        </div>
      ))}
      <div className={styles.legend}>
        {Object.entries(colorsMap).map(([value, { text, background }], index) => (
          <div
            key={index}
            className={styles.color}
            style={{ color: text, backgroundColor: background }}
          >
            {value}
          </div>
        ))}
      </div>
    </div>
  );
}

// solution for Ember usage
function TableChartCard({ icon, title, subTitle, width, height, ...rest }: CardWrapperProps & TableChartProps) {
  return (
    <CardWrapper {...{ icon, title, subTitle, width, height, className: styles.card }}>
      <TableChart {...rest} />
    </CardWrapper>
  );
}

export default TableChartCard;
