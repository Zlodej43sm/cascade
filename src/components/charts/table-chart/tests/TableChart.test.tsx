import { render } from '@testing-library/react';

import { cardArguments } from 'lib/components/card-wrapper/__mocks__/CardWrapper.mock';
import { chartArguments, randomColorArguments } from '../__mocks__/TableChart.mock';
import TableChartCard, { TableChart } from '../TableChart';

describe('TableChart', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<TableChart {...chartArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot with getColors', () => {
    const { asFragment } = render(<TableChart {...randomColorArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot for TableChartCard', () => {
    const { icon, title, subTitle } = cardArguments;
    const { asFragment } = render(<TableChartCard {...{ ...chartArguments, icon, title, subTitle }} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
