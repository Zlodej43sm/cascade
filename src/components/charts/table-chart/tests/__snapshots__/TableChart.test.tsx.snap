// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableChart Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid"
    data-testid="table-chart"
    style="--columns-count: 5; --rows-count: 6;"
  >
    <div
      class="row"
    >
      <div
        class="header"
      >
        Problem solving
      </div>
      <div
        class="header"
      >
        Job knowledge
      </div>
      <div
        class="header"
      >
        Time management
      </div>
      <div
        class="header"
      >
        Communication
      </div>
      <div
        class="header"
      >
        Achievement
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Ariel K."
          class="avatar"
          src="https://upload.wikimedia.org/wikipedia/commons/c/c0/Nicolas_Cage_Deauville_2013.jpg"
        />
        Ariel K.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
      >
        7
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
      >
        3
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
      >
        6
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Kate E."
          class="avatar"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQrHI7Ycu-v_h0GlkoMdN9JGhCmBK2w4uiCLw&usqp=CAU"
        />
        Kate E.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
      >
        2
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(122, 111, 226);"
      >
        9
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
      >
        7
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 235, 204);"
      >
        5
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
      >
        2
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Jerry V."
          class="avatar"
          src="https://hardtickettohomevideo.files.wordpress.com/2015/01/nick-cage-conair.jpg?w=570"
        />
        Jerry V.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
      >
        2
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(89, 75, 219);"
      >
        10
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
      >
        2
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
      >
        1
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
      >
        8
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Kevin H."
          class="avatar"
          src="https://s3-us-west-2.amazonaws.com/flx-editorial-wordpress/wp-content/uploads/2018/02/27181649/Nicolas-Cage-FFF.jpg"
        />
        Kevin H.
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
      >
        8
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
      >
        3
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
      >
        7
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
      >
        4
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Radha R."
          class="avatar"
          src="https://s3-us-west-2.amazonaws.com/flx-editorial-wordpress/wp-content/uploads/2018/09/11172859/Nicolas-Cage-Countdown-Rep.jpg"
        />
        Radha R.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
      >
        6
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
      >
        1
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
      >
        3
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
      >
        1
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
      >
        7
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Maat W."
          class="avatar"
          src="https://media.gq.com/photos/5582f8c409f0bee56440b484/master/pass/entertainment-2011-05-cage-match-nic-cage_300x430.jpg"
        />
        Maat W.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
      >
        6
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(89, 75, 219);"
      >
        10
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
      >
        8
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
      >
        2
      </div>
    </div>
    <div
      class="legend"
    >
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
      >
        1
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
      >
        2
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
      >
        3
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
      >
        4
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(255, 235, 204);"
      >
        5
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
      >
        6
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
      >
        7
      </div>
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
      >
        8
      </div>
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(122, 111, 226);"
      >
        9
      </div>
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(89, 75, 219);"
      >
        10
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TableChart Should match snapshot for TableChartCard 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-testid="card-wrapper-container"
  >
    <header
      class="header"
    >
      <div
        class="icon"
        data-testid="card-wrapper-icon"
      >
        <span>
          #
        </span>
      </div>
      <div>
        <div
          class="title"
          data-testid="card-wrapper-title"
        >
          FY21 Revenue
        </div>
        <div
          class="subTitle"
          data-testid="card-wrapper-subtitle"
        >
          As of today, in comparison with last month
        </div>
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="grid"
        data-testid="table-chart"
        style="--columns-count: 5; --rows-count: 6;"
      >
        <div
          class="row"
        >
          <div
            class="header"
          >
            Problem solving
          </div>
          <div
            class="header"
          >
            Job knowledge
          </div>
          <div
            class="header"
          >
            Time management
          </div>
          <div
            class="header"
          >
            Communication
          </div>
          <div
            class="header"
          >
            Achievement
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="name"
          >
            <img
              alt="Ariel K."
              class="avatar"
              src="https://upload.wikimedia.org/wikipedia/commons/c/c0/Nicolas_Cage_Deauville_2013.jpg"
            />
            Ariel K.
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
          >
            4
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
          >
            7
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
          >
            4
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
          >
            3
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
          >
            6
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="name"
          >
            <img
              alt="Kate E."
              class="avatar"
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQrHI7Ycu-v_h0GlkoMdN9JGhCmBK2w4uiCLw&usqp=CAU"
            />
            Kate E.
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
          >
            2
          </div>
          <div
            class="value"
            style="color: rgb(255, 255, 255); background-color: rgb(122, 111, 226);"
          >
            9
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
          >
            7
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 235, 204);"
          >
            5
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
          >
            2
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="name"
          >
            <img
              alt="Jerry V."
              class="avatar"
              src="https://hardtickettohomevideo.files.wordpress.com/2015/01/nick-cage-conair.jpg?w=570"
            />
            Jerry V.
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
          >
            2
          </div>
          <div
            class="value"
            style="color: rgb(255, 255, 255); background-color: rgb(89, 75, 219);"
          >
            10
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
          >
            2
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
          >
            1
          </div>
          <div
            class="value"
            style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
          >
            8
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="name"
          >
            <img
              alt="Kevin H."
              class="avatar"
              src="https://s3-us-west-2.amazonaws.com/flx-editorial-wordpress/wp-content/uploads/2018/02/27181649/Nicolas-Cage-FFF.jpg"
            />
            Kevin H.
          </div>
          <div
            class="value"
            style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
          >
            8
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
          >
            4
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
          >
            3
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
          >
            7
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
          >
            4
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="name"
          >
            <img
              alt="Radha R."
              class="avatar"
              src="https://s3-us-west-2.amazonaws.com/flx-editorial-wordpress/wp-content/uploads/2018/09/11172859/Nicolas-Cage-Countdown-Rep.jpg"
            />
            Radha R.
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
          >
            6
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
          >
            1
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
          >
            3
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
          >
            1
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
          >
            7
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="name"
          >
            <img
              alt="Maat W."
              class="avatar"
              src="https://media.gq.com/photos/5582f8c409f0bee56440b484/master/pass/entertainment-2011-05-cage-match-nic-cage_300x430.jpg"
            />
            Maat W.
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
          >
            6
          </div>
          <div
            class="value"
            style="color: rgb(255, 255, 255); background-color: rgb(89, 75, 219);"
          >
            10
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
          >
            4
          </div>
          <div
            class="value"
            style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
          >
            8
          </div>
          <div
            class="value"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
          >
            2
          </div>
        </div>
        <div
          class="legend"
        >
          <div
            class="color"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 153, 0);"
          >
            1
          </div>
          <div
            class="color"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 173, 51);"
          >
            2
          </div>
          <div
            class="color"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 194, 102);"
          >
            3
          </div>
          <div
            class="color"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 214, 153);"
          >
            4
          </div>
          <div
            class="color"
            style="color: rgb(0, 0, 0); background-color: rgb(255, 235, 204);"
          >
            5
          </div>
          <div
            class="color"
            style="color: rgb(0, 0, 0); background-color: rgb(222, 219, 248);"
          >
            6
          </div>
          <div
            class="color"
            style="color: rgb(0, 0, 0); background-color: rgb(189, 183, 241);"
          >
            7
          </div>
          <div
            class="color"
            style="color: rgb(255, 255, 255); background-color: rgb(155, 147, 233);"
          >
            8
          </div>
          <div
            class="color"
            style="color: rgb(255, 255, 255); background-color: rgb(122, 111, 226);"
          >
            9
          </div>
          <div
            class="color"
            style="color: rgb(255, 255, 255); background-color: rgb(89, 75, 219);"
          >
            10
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`TableChart Should match snapshot with getColors 1`] = `
<DocumentFragment>
  <div
    class="grid"
    data-testid="table-chart"
    style="--columns-count: 5; --rows-count: 6;"
  >
    <div
      class="row"
    >
      <div
        class="header"
      >
        Problem solving
      </div>
      <div
        class="header"
      >
        Job knowledge
      </div>
      <div
        class="header"
      >
        Time management
      </div>
      <div
        class="header"
      >
        Communication
      </div>
      <div
        class="header"
      >
        Achievement
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Ariel K."
          class="avatar"
          src="https://upload.wikimedia.org/wikipedia/commons/c/c0/Nicolas_Cage_Deauville_2013.jpg"
        />
        Ariel K.
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(66, 66, 66);"
      >
        7
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        3
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(20, 20, 20);"
      >
        6
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Kate E."
          class="avatar"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQrHI7Ycu-v_h0GlkoMdN9JGhCmBK2w4uiCLw&usqp=CAU"
        />
        Kate E.
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        2
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(158, 158, 158);"
      >
        9
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(66, 66, 66);"
      >
        7
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        5
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        2
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Jerry V."
          class="avatar"
          src="https://hardtickettohomevideo.files.wordpress.com/2015/01/nick-cage-conair.jpg?w=570"
        />
        Jerry V.
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        2
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(204, 204, 204);"
      >
        10
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        2
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        1
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(112, 112, 112);"
      >
        8
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Kevin H."
          class="avatar"
          src="https://s3-us-west-2.amazonaws.com/flx-editorial-wordpress/wp-content/uploads/2018/02/27181649/Nicolas-Cage-FFF.jpg"
        />
        Kevin H.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(112, 112, 112);"
      >
        8
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        3
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(66, 66, 66);"
      >
        7
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        4
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Radha R."
          class="avatar"
          src="https://s3-us-west-2.amazonaws.com/flx-editorial-wordpress/wp-content/uploads/2018/09/11172859/Nicolas-Cage-Countdown-Rep.jpg"
        />
        Radha R.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(20, 20, 20);"
      >
        6
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        1
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        3
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        1
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(66, 66, 66);"
      >
        7
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="name"
      >
        <img
          alt="Maat W."
          class="avatar"
          src="https://media.gq.com/photos/5582f8c409f0bee56440b484/master/pass/entertainment-2011-05-cage-match-nic-cage_300x430.jpg"
        />
        Maat W.
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(20, 20, 20);"
      >
        6
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(204, 204, 204);"
      >
        10
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        4
      </div>
      <div
        class="value"
        style="color: rgb(0, 0, 0); background-color: rgb(112, 112, 112);"
      >
        8
      </div>
      <div
        class="value"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        2
      </div>
    </div>
    <div
      class="legend"
    >
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        1
      </div>
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        2
      </div>
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        3
      </div>
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        4
      </div>
      <div
        class="color"
        style="color: rgb(255, 255, 255); background-color: rgb(0, 0, 0);"
      >
        5
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(20, 20, 20);"
      >
        6
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(66, 66, 66);"
      >
        7
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(112, 112, 112);"
      >
        8
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(158, 158, 158);"
      >
        9
      </div>
      <div
        class="color"
        style="color: rgb(0, 0, 0); background-color: rgb(204, 204, 204);"
      >
        10
      </div>
    </div>
  </div>
</DocumentFragment>
`;
